### ISSUE TITLE
[core] support primitive types in flags

### ISSUE THREAD

**Issue by BoscoJared** [BoscoJared/disparse#24]
*This feature / task write-up is in response to the original issue posted [here](https://github.com/BoscoJared/disparse/issues/22).*

**Feature Description**:  Currently, `@Flag` types can be put over the following types of fields: `Integer`, `Boolean`, `List`, and `String`.  In the case of `Integer` and `Boolean`, only the wrapper types are supported.  Disparse should be extending to be able to set the values of fields that are of type `int` or `boolean`.  After this feature, the total fields supported by `@Flag` will be `Integer`, `int`, `Boolean`, `boolean`, `List`, and `String`.

**Reproduce**:

1. Create a blank discord bot project
2. Create a class named `PrimitiveCommand.java`

```java
public class PrimitiveCommand {
    
    @ParsedEntity
    static class PrimitiveOpt {
        @Flag(shortName = 'n', longName = "number")
        int number = 0;
    }

    @ParsedEntity
    static class WrapperOpt {
        @Flag(shortName = 'n', longName = "number")
        Integer number = 0;
    }

    @CommandHandler(commandName = "test")
    public static DiscordResponse test(WrapperOpt option) {
        return DiscordResponse.of("You chose number:  " + option.number);
    }
}
```

Copy paste the above code into the class, resolving the imports as needed.  When `PrimitiveCommand::test` accepts `WrapperOpt` the command will work because it is using the wrapper `Integer` type.  When `PrimitiveCommand::test` is switched to use `PrimitiveOpt` the command will throw a stacktrace due to not yet supporting the `int` primitive type in Disparse.

**Modules Affected**:  This is a core issue only, and therefore changes should only need to be made into `disparse-core`.  The commit message should following the contributing guidelines and be something like: `[core] support primitive types as flag fields`.

**Where to look**: [Utils.java](https://github.com/BoscoJared/disparse/blob/6b97b4e45c36b61704d3a5f7bdace4c8bb91d140/disparse-core/src/main/java/disparse/parser/reflection/Utils.java#L28) has a few conditions explicitly type checking the field and setting to an enum.  This would need to be updated first.

[CommandRegistrar.java](https://github.com/BoscoJared/disparse/blob/master/disparse-core/src/main/java/disparse/parser/dispatch/CommandRegistrar.java#L363) is where the command will eventually be dispatched with the correctly set values.  This may or may not need to change after Utils.java is changed to accept both Integers and ints.  

**Tests**:  Ideally a PR would include a few tests in [DispatchIntegrationTests.java](https://github.com/BoscoJared/disparse/blob/master/disparse-core/src/test/java/disparse/discord/DispatchIntegrationTests.java) but ultimately it is not required.

As always, read the [contributing guidelines](https://github.com/BoscoJared/disparse/blob/master/CONTRIBUTING.md) and/or reach out on Discord @Bosco#8564 for help or discussion.

**Comment by johan123456718** [comment]
I can do this ticket

**Comment by BoscoJared** [comment]
Closed by #25
